okhttp/src/main/java/okhttp3/Response.kt

```diff
@@ -302,6 +302,21 @@ class Response internal constructor(
     checkNotNull(body) { "response is not eligible for a body and must not be closed" }.close()
   }
 
+  /**
+   * Returns the sequence of Responses that led to this final Response.
+   */
+  fun responses(): Iterable<Response> = generateSequence(this) {
+    it.priorResponse
+  }.asIterable()
+
+  /**
+   * Returns the number of network Responses that were received to get to this final Response.
+   *
+   * The minimum is 1. A 401 response that is then correctly authenticated will have a response
+   * count of 2.
+   */
+  val responseCount = responses().count()
```

Comment:
I know it’s not as critical anymore, but I don’t love the extra allocations this creates in every Response creation.

Reply:
This is a great point, I would definitely leave this as a function.